cl/search/models.py

```diff
@@ -1326,6 +1332,11 @@ class RECAPDocument(AbstractPacerDocument, AbstractPDF, AbstractDateTimeModel):
         ),
         blank=True,
     )
+    acms_document_guid = models.CharField(
+        help_text="The GUID of the document in ACMS.",
+        max_length=64,
+        blank=True,
+    )
```

Comment:
Shouldn't we have a unique constraint that covers `acms_document_guid` in some way to prevent duplicates?

Reply:
I think it may be better to use nulls instead of allowing empty strings here as I don't think multi-column unique constraints would work correctly otherwise for cases where `acms_document_guid` is not used.

Reply:
@johnhawkinson or @albertisfu, either of you have opinions about this?

Reply:
We currently have the following multi-column constraint on `RECAPDocument`:

```
unique_together = (
            "docket_entry",
            "document_number",
            "attachment_number",
        )
```

I believe this constraint will also help prevent duplicates in ACMS attachments.

The `acms_document_guid` will store the `docketDocumentDetailsId`, which can only be found on the ACMS attachment page and the download confirmation page, so is not available for entries in the docket report, so we cannot use it to prevent duplicates in the main documents only to prevent duplicated attachments.

![Screenshot 2023-12-07 at 9 43 22](https://github.com/freelawproject/courtlistener/assets/486004/aecc1d9a-da73-4684-97b9-cbc65700cbd2)

We also get the "attachment_number" in the ACMS attachment page, which is the `documentNumber` in the previous screenshot, so unless it's possible to have multiple attachments with the same `attachment_number` but different `acms_document_guid` I think including `acms_document_guid`  to the multi-column constraint is not required.